### PR TITLE
[Merged by Bors] - chore(ModelTheory/ElementarySubstructures): Split elementary substructures into their own file

### PR DESCRIPTION
**Pull request by awainverse**
This PR splits the file `ModelTheory/ElementaryMaps` into two files, moving elementary substructure code into `ModelTheory/ElementarySubstructures`, to make room for new API on those.

Two basic lemmas, `FirstOrder.Language.Substructure.realize_boundedFormula_top` and `FirstOrder.Language.Substructure.realize_formula_top`, are instead moved to the file `ModelTheory/Substructures`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Group/Arithmetic.lean [hunk 1]
```diff
@@ -524,6 +523,11 @@ instance (priority := 100) measurableDiv₂_of_mul_inv (G : Type*) [MeasurableSp
 #align has_measurable_div₂_of_mul_inv measurableDiv₂_of_mul_inv
 #align has_measurable_div₂_of_add_neg measurableDiv₂_of_add_neg
 
+-- See note [lower instance priority]
+instance (priority := 100) MeasurableDiv.toMeasurableInv [MeasurableSpace α] [Group α]
+    [MeasurableDiv α] : MeasurableInv α where
+  measurable_inv := by simpa using measurable_const_div (1 : α)
+
```
Comment (eric-wieser) [inline Mathlib/MeasureTheory/Group/Arithmetic.lean:530]: These look unrelated to the current PR; perhaps merging master will make them go away?

Comment (awainverse) [reply Mathlib/MeasureTheory/Group/Arithmetic.lean]: Yes, thanks. Now only model theory files are affected.

File: Mathlib/ModelTheory/ElementarySubstructures.lean [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+# Elementary Substructures
+
+## Main Definitions
+* A `FirstOrder.Language.ElementarySubstructure` is a substructure where the realization of each
+  formula agrees with the realization in the larger model.
+
+## Main Results
+* The Tarski-Vaught Test for substructures:
+`FirstOrder.Language.Substructure.isElementary_of_exists` gives a simple criterion for a
+substructure to be elementary.
```
Comment (eric-wieser) [inline Mathlib/ModelTheory/ElementarySubstructures.lean:18]:
```suggestion
* The Tarski-Vaught Test for substructures:
  `FirstOrder.Language.Substructure.isElementary_of_exists` gives a simple criterion for a
  substructure to be elementary.
```

File: Mathlib/ModelTheory/ElementarySubstructures.lean [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+#align first_order.language.elementary_substructure.subtype FirstOrder.Language.ElementarySubstructure.subtype
+
+@[simp]
+theorem coeSubtype {S : L.ElementarySubstructure M} : ⇑S.subtype = ((↑) : S → M) :=
+  rfl
+#align first_order.language.elementary_substructure.coe_subtype FirstOrder.Language.ElementarySubstructure.coeSubtype
+
+/-- The substructure `M` of the structure `M` is elementary. -/
+instance instTop : Top (L.ElementarySubstructure M) :=
+  ⟨⟨⊤, fun _ _ _ => (Substructure.realize_formula_top _ _).symm⟩⟩
```
Comment (eric-wieser) [inline Mathlib/ModelTheory/ElementarySubstructures.lean:106]: This proof grew two underscores; was that deliberate, or should the arguments to `realize_formula_top` have been kept implicit?

If it was deliberate, please put it in the PR description.

bors d+

Comment (awainverse) [reply Mathlib/ModelTheory/ElementarySubstructures.lean]: It was not deliberate - I have made those variables implicit again.
In fact, I have now moved those two lemmas to the Substructures file, as I think they really belong there. I will adjust the PR description accordingly.
